contracts/JBETHERC20SplitsPayer.sol

```diff
@@ -0,0 +1,412 @@
+        IERC20(_token).transferFrom(msg.sender, payable(address(this)), _amount);
+    } else {
+      _payer = address(this);
+      _amount = msg.value;
+      _decimals = 18;
+    }
+
+    // Pay the split and get a reference to the amount paid.
+    uint256 _leftoverAmount = _payToSplits(
+      defaultProjectId,
+      defaultSplitsDomain,
+      defaultSplitsGroup,
+      _token,
+      _payer,
+      _amount,
+      _decimals
+    );
+
+    // If there is no leftover amount, nothing left to pay.
+    if (_leftoverAmount == 0) return;
+
+    // If there's a default project ID, try to add to its balance.
+    if (_projectId != 0)
+      // Add to the project's balance.
+      _addToBalance(_projectId, _token, _payer, _leftoverAmount, _decimals, _memo);
+
+      // Otherwise, send a payment to the beneficiary.
+    else
+      Address.sendValue(
+        defaultBeneficiary != address(0) ? defaultBeneficiary : payable(msg.sender),
+        _leftoverAmount
+      );
+  }
+
+  /** 
+    @notice 
+    Split the contract's balance between all splits.
+
+    @param _splitsProjectId The ID of the project to which the splits belong.
+    @param _splitsDomain The splits domain to which the group belongs.
+    @param _splitsGroup The splits group to pay.
+    @param _token The token being paid in.
+    @param _payer The address from whom the payment is originating.
+    @param _amount The amount of tokens being paid, as a fixed point number. If this terminal's token is ETH, this is ignored and msg.value is used in its place.
+    @param _decimals The number of decimals in the `_amount` fixed point number. 
+  */
+  function _payToSplits(
+    uint256 _splitsProjectId,
+    uint256 _splitsDomain,
+    uint256 _splitsGroup,
+    address _token,
+    address _payer,
+    uint256 _amount,
+    uint256 _decimals
+  ) internal virtual returns (uint256 leftoverAmount) {
+    // Get a reference to the item's settlement splits.
+    JBSplit[] memory _splits = splitsStore.splitsOf(_splitsProjectId, _splitsDomain, _splitsGroup);
+
+    // Set the leftover amount to the initial balance.
+    leftoverAmount = _amount;
+
+    // Settle between all splits.
+    for (uint256 i = 0; i < _splits.length; i++) {
+      // Get a reference to the split being iterated on.
+      JBSplit memory _split = _splits[i];
+      // Pay the split and get a reference to the amount paid.
+      // The amount to send towards the split.
+      uint256 _splitAmount = PRBMath.mulDiv(
+        _amount,
+        _split.percent,
+        JBConstants.SPLITS_TOTAL_PERCENT
+      );
+
+      if (_splitAmount > 0) {
+        // Transfer tokens to the split.
+        // If there's an allocator set, transfer to its `allocate` function.
+        if (_split.allocator != IJBSplitAllocator(address(0))) {
+          // Create the data to send to the allocator.
+          JBSplitAllocationData memory _data = JBSplitAllocationData(
+            _payer,
+            _splitAmount,
+            0,
```

Comment:
Shouldn't we pass the decimal? One allocator can have multiple splitsPayer/multiple token, no?

Reply:
we should def pass the decimal.